packages/nx/src/tasks-runner/task-orchestrator.ts

```diff
@@ -376,20 +383,69 @@ export class TaskOrchestrator {
 
     // the task wasn't cached
     if (results.length === 0) {
-      // cache prep
-      const { code, terminalOutput } = await this.runTaskInForkedProcess(
+      const shouldPrefix =
+        streamOutput && process.env.NX_PREFIX_OUTPUT === 'true';
+      const targetConfiguration = getTargetConfigurationForTask(
         task,
-        env,
-        pipeOutput,
-        temporaryOutputPath,
-        streamOutput
+        this.projectGraph
       );
+      if (targetConfiguration.executor === 'nx:run-commands' && !shouldPrefix) {
```

Comment:
Add an environment variable, `NX_RUN_COMMANDS_DIRECTLY=true`, to opt in for now.

We'll remove it once we're sure it works.